src/main.rs

```diff
@@ -79,14 +83,14 @@ fn main() -> Result<(), Box<dyn Error>> {
     let matches = match opts.parse(std::env::args()) {
         Ok(it) => it,
         Err(err) => {
-            usage("commit-analzer", opts);
+            usage(opts);
             return Err(err.into());
         }
     };
-    if matches.free.len() < 2 {
-        usage(&matches.free[0], opts);
+    if matches.opt_present("h") || matches.free.len() < 0x2 {
+        usage(opts);
         return Ok(());
-    }
+    };
```

Comment:
Also a habit from C / C++.  I always set all semicolons where they are allowed and not linted away.

Many C / C++ software engineers probably know that:  they specify a new type (both `struct` and `class`) together with all associated methods / member functions / utility functions, but there are some smaller bugs due to which the compiler denies the compilation.  Okay, one fixes them one after another but still, the new type does not compile:  one forgot to end the `class` / `struct` definition with a semicolon.

If you always set just the required amount of semicolons, you need to learn by heart when you need one.  The other way around, when you always set all semicolons, even those which are obsolete, you need to learn where to not set one.

It turns out that learning the exceptions is more effective than learning all places you explicitly need one because there are usually more required places than places where you should omit the semicolon.

If you dislike it, I can remove it.

Reply:
Hmm, for one, you can always insert almost everywhere semicolon. So would you write also such c code?
```c
#include<stdlib.h>;
;
void foo(){};
```

And secondly, this is kind of part of learning a programming language. Knowing how it works and why it works, this would also includes semicolons. 😅

Reply:
First, the shown code is invalid according to `-std=c99`.  "ISO C denies semicolons outside of functions" would be the answer of the compiler to that, at least GCC thinks that way.  But basically, yes, this looks like good C code to me.

Second, did you read "Crafting Interpreters" by Robert Nystrom?  (https://github.com/munificent/craftinginterpreters)  He writes that similarities between coding languages, especially when designing a new one, help to reduce the learning effort.  And to some point, Rust looks quite like C++, so basically, knowing C++ and how to apply its semicolon rules should suffice to deduce the basics of how Rust should handle instruction delimitation.

Reply:
Due to this discussion on that single semicolon, I assume I should rather remove it, right?

Reply:
> Due to this discussion on that single semicolon, I assume I should rather remove it, right?

Yeah 👍

Reply:
Done.

I wonder why `rustfmt` alias `cargo fmt` does not lint such things away.  Maybe a candidate for a Feature Request.

Reply:
> Second, did you read "Crafting Interpreters" by Robert Nystrom? (https://github.com/munificent/craftinginterpreters) He writes that similarities between coding languages, especially when designing a new one, help to reduce the learning effort. And to some point, Rust looks quite like C++, so basically, knowing C++ and how to apply its semicolon rules should suffice to deduce the basics of how Rust should handle instruction delimitation.

I did indeed. But this should not stop one from learning the other language completely. But I'm not sure, if I agree completely, e.g. enums are quite different between c++ and rust, not only on the syntactic level, but also there. I think it should make reading a foreign programming language easier, but I don't think it should affect how you write code in the long term. 😅